xxxx-s-latency-random.md

```diff
@@ -0,0 +1,61 @@
+
+The Helium network is an IOT network composed of several nodes, including hotspots.  
+Hotspots bring value to the IOT network in routing data packets.
+
+It can be inferred that the value they provide is proportional to the RF coverage they have.
+
+Thus, Helium implemented a Proof of Coverage mechanism where some hotspots would broadcast a beacon, and neighbors would confirm they saw the message.
+
+Hotspot owners are rewarded in crypto tokens whenever their hotspot(s) route data packets, broadcast a beacon, or witness a beacon. This is the value they bring to the network.
+
+## Current state
+
+Today, this value is rewarded in tokens based on latency, with an emphasis of the importance of a few milliseconds.
+
+## Latency in an IOT network
+
+Helium is an IOT network based on LoRaWAN.
+
+LoRaWAN supports both ways communication, and support acks (confirmed uplinks/downlinks). Here is the most common delays in the world:
+
+| Situation                                          | Minimal latency (ms) | Optimal latency (ms) |
+|----------------------------------------------------|----------------------|----------------------|
+| Unconfirmed uplink, class A, no scheduled downlink | none                 | 1000                 |
+| Unconfirmed uplink, class A, scheduled downlink    | 2000                 | 1000                 |
+| Confirmed uplink                                   | 2000                 | 1000                 |
+| Join Request                                       | 6000                 | 5000                 |
+
+For any uplink, there may be some considerations on the use case. For example, plotting the temperature of a refrigerator for historical data and a school panic button do not have the same requirements.  
+The former would tolerate minutes if not hours of delay in data collection (as long as the data is there eventually), however the latter would not tolerate a latency over a few seconds.
+
+Thus, LoRaWAN has a latency requirement in the order of magnitude of about 10^3 milliseconds.
+
+## Latency vs. value
+
+After looking at the value provided by hotspots and how they are rewarded, it appears that there are several orders of magnitude of difference between what is rewarded and what is actually useful to the network (10^1 vs 10^3 milliseconds).
+
+## Proposition
+
+Ideally, we should sort hotspots that perform the work (witnessing or routing a packet) using a KPI that fairly represents the value they bring to the network.
+
+As a reminder of said value from the first paragraph:
+
+* routing packets: unless they are unreasonably slow (see table above), all hotspots provide the same amount and value of work (e.g. routing the packet from A to B)
+* witnessing coverage: the uptime, coverage, RF performance, reliability, power usage, data consumption, etc.
+
+It is very hard to assert a reliable KPI to assess this value.  
+Today, latency fails to do so.
+
+Short of a better KPI, randomness seems the fairest option.
```

Comment:
Would one option be to actually use the above "at least this fast" as a minimum (i.e. KPI!) and then randomly reward within that set? vs full random 14?

Reply:
I have no opinion on 14 per se, or 14 vs everybody. Thus, I'd like this to be out of the discussion (e.g. keep the current 14).

Indeed, "at least this fast" is a good question. As highlighted in the table, the "at least" is 2000ms if there is a class A downlink scheduled, and up to the application layer if there is not (not less than a few seconds even for the most dramatic use case).

Thus, depending on the downlink queue, the threshold could be 2000ms or say 5000ms if empty queue.

However, the backhaul latency isn't the only metric end-to-end, so headroom must be added and processing time given to the different Helium cloud nodes.

If I were to pick the threshold, I'd say 800ms RTT if there is a downlink, and 2500ms if there is none.

Reply:
Strictly speaking, there is:

1. uplink latency
2. cloud processing
3. downlink latency

At step 2 we know what the downlink time budget left is (RX window chosen, processing done). Thus we can at this step (with an additional processing headroom) select which hotspots actually could be chosen for a downlink (with an additional headroom to accommodate the downlink latency value and variance).

Hotspots could be eliminated before step 3 by assuming that the latency is symmetric.

(only if there is a scheduled downlink, otherwise the threshold could be relaxed to the 2500ms I mentioned above)

Reply:
right.. (and yes I wasn't picky about the number of hotspots to be rewarded that fall inside the "fast enough" line). 

Would be good to work what you said into the proposal, since it actually adds some initial detail around the max latency value (or whatever the right name is)

Reply:
will do

Reply:
done, is that better?